fe/src/main/java/org/apache/doris/load/DeleteHandler.java

```diff
@@ -0,0 +1,684 @@
+        queue = new LinkedBlockingQueue(Config.delete_thread_num);
+        lock = new ReentrantReadWriteLock(true);
+        // start checker
+        checker = new DeleteTaskChecker(queue);
+        checker.start();
+    }
+
+    public void process(DeleteStmt stmt) throws DdlException {
+        String dbName = stmt.getDbName();
+        String tableName = stmt.getTableName();
+        String partitionName = stmt.getPartitionName();
+        List<Predicate> conditions = stmt.getDeleteConditions();
+        Database db = Catalog.getInstance().getDb(dbName);
+        if (db == null) {
+            throw new DdlException("Db does not exist. name: " + dbName);
+        }
+
+        DeleteTask deleteTask = null;
+        DeleteInfo deleteInfo = null;
+        long transactionId;
+        db.readLock();
+        try {
+            Table table = db.getTable(tableName);
+            if (table == null) {
+                throw new DdlException("Table does not exist. name: " + tableName);
+            }
+
+            if (table.getType() != Table.TableType.OLAP) {
+                throw new DdlException("Not olap type table. type: " + table.getType().name());
+            }
+            OlapTable olapTable = (OlapTable) table;
+
+            if (olapTable.getState() != OlapTable.OlapTableState.NORMAL) {
+                throw new DdlException("Table's state is not normal: " + tableName);
+            }
+
+
+            if (partitionName == null) {
+                if (olapTable.getPartitionInfo().getType() == PartitionType.RANGE) {
+                    throw new DdlException("This is a range partitioned table."
+                            + " You should specify partition in delete stmt");
+                } else {
+                    // this is a unpartitioned table, use table name as partition name
+                    partitionName = olapTable.getName();
+                }
+            }
+
+            Partition partition = olapTable.getPartition(partitionName);
+            if (partition == null) {
+                throw new DdlException("Partition does not exist. name: " + partitionName);
+            }
+
+            List<String> deleteConditions = Lists.newArrayList();
+
+            // pre check
+            checkDeleteV2(olapTable, partition, conditions, deleteConditions, true);
+
+            // generate label
+            String label = "delete_" + UUID.randomUUID();
+
+            // begin txn here and generate txn id
+            transactionId = Catalog.getCurrentGlobalTransactionMgr().beginTransaction(db.getId(),
+                    Lists.newArrayList(table.getId()), label,"FE: " + FrontendOptions.getLocalHostAddress(),
+                    TransactionState.LoadJobSourceType.FRONTEND, Config.stream_load_default_timeout_second);
+
+            TransactionState state = Catalog.getCurrentGlobalTransactionMgr()
+                    .getTransactionState(transactionId);
+            if (state == null) {
```

Comment:
If begin transaction failed, exception will be thrown. So no need to check here.

Reply:
I've deleted it